### PR TITLE
Spark 3.5: Fix testDeleteFileThenMetadataDelete failure due to table not refreshed

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java [hunk 1]
```diff
@@ -166,6 +166,28 @@ public static Object[][] parameters() {
         false,
         "test",
         DISTRIBUTED
+      },
+      {
+        "spark_catalog",
```
Comment (manuzhang) [inline spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java:171]: `TestDelete.testDeleteFileThenMetadataDelete` will fail for this catalog without table refresh below.

Comment (aokolnychyi) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java:171]: Our DELETE, UPDATE, and MERGE tests are probably the longest running tests in the project, which are already tested with 4 catalog configs. Could you elaborate on why it is important to cover this case as well?

Comment (manuzhang) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java:171]: Yet we still haven't covered all file formats and other combinations of configurations (e.g. `cache-enabled`). In this test, `SparkSessionCatalog` is skipped due to its `avro` file format. I happened to find this test could fail due to table not being refreshed if not skipped.

Maybe there is a better way to cover more cases?

Comment (manuzhang) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java:171]: @aokolnychyi any more thoughts?

**Pull request closed without merging**